tests/upload_test.go

```diff
@@ -183,28 +200,87 @@ func startUploadProxyPortForward(f *framework.Framework) (string, *exec.Cmd, err
 	return url, cmd, nil
 }
 
-func uploadImage(portForwardURL, token string, expectedStatus int) error {
-	url := portForwardURL + "/v1alpha1/upload"
+func formRequestFunc(url string) (*http.Request, error) {
+	f, err := os.Open(utils.UploadFile)
+	if err != nil {
+		return nil, err
+	}
+
+	pipeReader, pipeWriter := io.Pipe()
+	multipartWriter := multipart.NewWriter(pipeWriter)
 
+	req, err := http.NewRequest("POST", url, pipeReader)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Content-Type", multipartWriter.FormDataContentType())
+
+	go func() {
+		defer GinkgoRecover()
+
+		formFile, err := multipartWriter.CreateFormFile("file", utils.UploadFile)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = io.Copy(formFile, f)
+		Expect(err).ToNot(HaveOccurred())
+
+		multipartWriter.Close()
```

Comment:
Should these closes be deferred, if io.Copy fails, they are never closed.

Reply:
not that it matters much in test scenereo, but I deferred closing the file an pie.  But the multipart writer should only be closed when `io.Copy` succeeded.  And it is probably smart to check the return there